app/components/apply-blob-container-legal-hold/variable.tf

```diff
@@ -0,0 +1,9 @@
+variable "blob_store_account_container_pairs" {
+  description = "A list of storage account name and container pairs"
+  type        = list(list(string))
```

Comment:
This would be clearer as an object:

`list(object({account = string, container = string}))`

Any reason we can't use that?